vcc/include/shady.h

```diff
@@ -48,11 +48,21 @@ typedef __attribute__((address_space(0x1001))) struct __shady_builtin_sampler2D*
 typedef __attribute__((address_space(0x1002))) struct __shady_builtin_sampler3D* sampler3D;
 typedef __attribute__((address_space(0x1003))) struct __shady_builtin_sampler3D* samplerCube;
 
+typedef __attribute__((address_space(0x1100))) struct __shady_builtin_image2D *image1D;
+typedef __attribute__((address_space(0x1101))) struct __shady_builtin_image2D *image2D;
+typedef __attribute__((address_space(0x1102))) struct __shady_builtin_image2D *image3D;
+typedef __attribute__((address_space(0x1103))) struct __shady_builtin_image2D *imageCube;
+
 native_vec4 texture1D(const sampler1D, float)           __asm__("shady::impure_op::spirv.core::87::Invocation");
 native_vec4 texture2D(const sampler2D, native_vec2)     __asm__("shady::impure_op::spirv.core::87::Invocation");
 native_vec4 texture3D(const sampler3D, native_vec3)     __asm__("shady::impure_op::spirv.core::87::Invocation");
 native_vec4 textureCube(const samplerCube, native_vec3) __asm__("shady::impure_op::spirv.core::87::Invocation");
 
+native_ivec2 imageSize(image2D img) __asm__("shady::pure_op::spirv.core::104::Invocation");
```

Comment:
this PR would make a lot more sense if this also included the load ops

Reply:
@Hugobros3 my bad, I realized looking at the rest of the header file I realized I hadn't done a very good job here. I hope the changes I have made are an improvement.